model/jpa/src/main/java/org/keycloak/models/jpa/JpaUserProvider.java

```diff
@@ -105,7 +105,9 @@ public UserModel addUser(RealmModel realm, String id, String username, boolean a
 
         UserEntity entity = new UserEntity();
         entity.setId(id);
-        entity.setCreatedTimestamp(System.currentTimeMillis());
+        long now = System.currentTimeMillis();
+        entity.setCreatedTimestamp(now);
+        entity.setLastUpdatedTimestamp(now);
```

Comment:
how about setting the lastUpdate timestamp to null here and really only update it on a real change?

Reply:
Having lastUpdated already set on creation has the advantage that it is never null (at least for users created after introduction of this feature).
Furthermore, it has the advantage that readers of a user representation just have to check one timestamp in order to find out when the data was changed/created. (Otherwise they would have to check both created and lastUpdated, which would be less convenient imho.)

Reply:
This makes sense to me and I think your implementation is fine.
However having a dedicated null value would make it easier to check if the user has been updated at all or not, but since this also possible with your change by comparing both timestamps, it should be ok.

Out of couriosity:
Have you looked at how often this update timestamp is updated during normal API interactions?
- Create user?
- Update user directly (enable / disable / change profile field)?
- Update user attribute?
- Change user group / role assignment?

Reply:
The current implementation should be aligned with Keycloak Events. In some cases, for example when changing a user via the Admin REST API (Admin Event UPDATE), there might be no "real changes" - but it would be confusing when users receive an UPDATE event, but there is no change to the lastUpdatedTimestamp.